src/components/ModalSendTx.js

```diff
@@ -65,67 +60,49 @@ class ModalSendTx extends React.Component {
       if (this.sendErrorMessage && this.props.onSendError) {
         // If had an error sending and have an error method
         this.props.onSendError(this.sendErrorMessage);
-        this.setState({ step: 0 });
-        return;
       }
-
-      this.setState({ errorMessage: '', step: 0 }, () => {
-        this.refs.pinInput.refs.pin.value = '';
-      });
     });
 
-    $('#pinModal').on('shown.bs.modal', (e) => {
-      this.refs.pinInput.refs.pin.focus();
-    });
+    // Start the promise and ignore its results
+    this.processTxWithPin(this.props.pin);
   }
 
   componentWillUnmount = () => {
     // Removing all event listeners
-    $('#pinModal').off();
-    $('#pinModal').modal('hide');
+    $('#sendTxModal').off();
+    $('#sendTxModal').modal('hide');
   }
 
   /**
    * Method called after user clicks the 'Go' button. We validate the form and that the pin is correct, then call a method from props
    *
-   * @param {Object} e Event emitted when button is clicked
+   * @param {string} pin Pin received from the user
    */
-  handlePin = async (e) => {
-    e.preventDefault();
-    if (this.refs.formPin.checkValidity() === false) {
-      this.refs.formPin.classList.add('was-validated');
+  processTxWithPin = async (pin) => {
+    $('#sendTxModal').data('bs.modal')._config.backdrop = 'static';
+    $('#sendTxModal').data('bs.modal')._config.keyboard = false;
+
+    // If we are using the wallet service facade, we should avail of the validated PIN
+    // to renew the auth token.
+    if (this.props.useWalletService) {
+      await this.props.wallet.validateAndRenewAuthToken(pin);
+    }
+
+    this.sendTransaction = await this.props.prepareSendTransaction(pin);
```

Comment:
This is actually done immediately when the modal is mounted: `processTxWithPin` is called at the `componentDidMount` function.

I just left it at a separate function from the `componentDidMount` to avoid making it `async` and maybe breaking some _Bootstrap_ component expectations. And I think it helped a bit with the readability also.

The `wallet.validateAndRenewAuthToken()` is also important to be executed before the `prepareSendTransaction()` callback, as their implementations can use the wallet service connection.

Reply:
Yeah, I agree, however you are setting `this.setState({ loading: true });` only after the first request finishes. In that case, we wouldn't see any loading on the screen. 

My comment was about showing the loading icon at the start of the `processTxWithPin` method.

Reply:
Good point, a little more refactoring was necessary to do this change on [91b7e14](https://github.com/HathorNetwork/hathor-wallet/pull/392/commits/91b7e14976340dc6c72734dbb6145bc680790fc3), since the `loading` state was used to trigger updates for other non-state properties.

Also added some unit tests on [0624d2a](https://github.com/HathorNetwork/hathor-wallet/pull/392/commits/0624d2a954873d2be66e3513e8f90eed70128765) since it was getting hard to read the diffs.